apps/mull-api/src/app/event/event.service.ts

```diff
@@ -47,33 +46,40 @@ export class EventService {
       .getMany();
   }
 
-  /**
-   * Find all the events that the user has not joined, created or is not co-hosting
-   * @param userId
-   */
-  findDiscoverEvent(userId: number): Promise<Event[]> {
+  findJoinedEvent(eventId: number, userId: number): Promise<Event> {
     return this.eventRepository
       .createQueryBuilder('event')
+      .leftJoin('event.participants', 'user')
       .leftJoin('event.coHosts', 'coHost')
-      .leftJoin('event.participants', 'participant')
       .leftJoin('event.host', 'host')
       .where(
         new Brackets((qb) => {
-          qb.where('coHost.id != :userId', { userId }).orWhere('coHost.id IS NULL');
-        })
-      )
-      .andWhere(
-        new Brackets((qb) => {
-          qb.where('participant.id != :userId', { userId }).orWhere('participant.id IS NULL');
+          qb.where('coHost.id = :userId', { userId })
+            .orWhere('user.id = :userId', { userId })
+            .orWhere('host.id = :userId', { userId });
```

Comment:
I'm a little confused, how come we're checking if the host has joined the event, or am I miss interpreting something?

Reply:
I think the event created by a user should be queried separately since it will belong in its own tab in the homepage and have unique logic such as cancelling events, upgrading users to coHosts and the ability to remove participants.

Reply:
Right, the query shouldn't show the host, since it should show 'cancel event' instead for those users, will remove the host from this query. For coHost, would need to re-affirm with the team if they can leave or actually cancel the event as well